packages/clay-drop-down/src/Item.tsx

```diff
@@ -15,6 +15,11 @@ interface IProps
 	 */
 	active?: boolean;
 
+	/**
+	 * Span component to render. It can be a one component that will replace the markup.
+	 */
+	component?: 'span' | React.ComponentType<any>;
```

Comment:
I think this API might be a bit overkill for clay's use case. We coudl probably get away with using only an anchor or a  button.

Reply:
Hmm, maybe it is, I think to cover the case above, I think we can remove the `ItemElement` wrapper and let those who use the low level component decide to use a `a`, `button` element... via `children `, I think it is more flexible so, when the element is not passed the classNames (`dropdown-item`) can live in `li`. What do you think?

Reply:
Yeah that sounds good to me

Reply:
I think it will not work very well 🙂 so that it works well I would have to remove the icons from the `Item` because the icons have to live inside the elements, be it `a`, `button` or whatever. I think I will keep it as is and remove the `component` API I will try to modify cover the above use case by adding another new API to add ref to the anchor or button.